src/engine/sparqlExpressions/NaryExpression.h

```diff
@@ -167,20 +169,63 @@ using LatitudeExpression =
 using DistExpression =
     NARY<2, FV<decltype(ad_utility::wktDist), StringValueGetter>>;
 
+// Date functions.
+template <size_t pos1, size_t pos2>
+inline auto extractNumberFromDate = [](const auto& dateAsString) -> long int {
+  static_assert(pos2 > pos1);
+  const size_t offset = std::strlen(VALUE_DATE_PREFIX) + 15;
+  if (dateAsString.size() >= pos2) {
+    return std::atol(dateAsString.data() + offset + pos1);
+  } else {
+    return 0;
+  }
+};
+using YearExpression =
+    NARY<1, FV<decltype(extractNumberFromDate<0, 4>), StringValueGetter>>;
+using MonthExpression =
+    NARY<1, FV<decltype(extractNumberFromDate<5, 7>), StringValueGetter>>;
+using DayExpression =
+    NARY<1, FV<decltype(extractNumberFromDate<8, 10>), StringValueGetter>>;
+
+// String functions. NOTE: The implementation assumes that if the string
+// contains one quote, it contains at least two.
+inline auto str = [](const auto& s) -> std::string { return s; };
+using StrExpression = NARY<1, FV<decltype(str), StringValueGetter>>;
+
+// String functions. NOTE: The implementation assumes that if the string
+// contains one quote, it contains at least two.
+inline auto strlen = [](const auto& s) -> long int {
+  auto pos = s.find_last_of('"');
+  if (pos == std::string::npos || pos == 0) {
+    return static_cast<long int>(s.size());
+  } else {
+    return static_cast<long int>(pos - 1);
+  }
+};
+using StrlenExpression = NARY<1, FV<decltype(strlen), StringValueGetter>>;
+
```

Comment:
I think the length of this header is getting a little bit out of control. Maybe we should have 
Headers "StringFunctions.h" and `NumericFunctions.h`, `DateFunctions.h` etc. with the corresponding `.cpp` files
(This is something, that you could do for this PR).

Also we could maybe avoid having the lambdas all in the header, since the rest of the definitions is in the .cpp-File (for compilation time reasons[maybe not that relevant anymore]) and for readability of the headers [very relevant]). 
But this is probably something that I should tackle myself.